Add a link explaning the guard directive used in the code samples (#614)
vaadin/docs: articles/ds/components/dialog/index.asciidoc

----
--

[IMPORTANT]

> The typescript examples in this page use an inline `.renderer` function, which is encapsulated by the `guard` directive in Lit. This is done for performance reasons. For more details on the `guard` directive, please refer to the https://lit.dev/docs/templates/directives/#guard[official documentation].

Reviewer: The content of the asciidoc should be relevant to the Design System and ideally language independent. Using Lit guards is a technical detail in some examples so this notice would work better as a code comment next to where the `guard` is used.

In future, we'll replace the `guard` directives in the examples with something similar to https://github.com/web-padawan/lit-vaadin-helpers

Reviewer 2: You can have language-specific notes also in example blocks for each language. Shouldn't use admonitions there though.

I made some language fixes, but maybe this issue should be solved.

Author: How do I add notes to the example blocks? I seems to be just linking to an external file.

Reviewer 2: You can have description text above the `[source, Java]` line (leave one empty line between the text). This works inside `[.example]` boxes. Each language (group) can have such description. See for example the code example in https://vaadin.com/docs/latest/flow/application/routing-navigation

> The typescript examples in this page use an inline `.renderer` function, which is encapsulated by the `guard` directive in Lit. This is done for performance reasons. For more details on the `guard` directive, please refer to the https://lit.dev/docs/templates/directives/#guard[official documentation].

Reviewer 2: ```suggestion
The TypeScript examples in this article use an inline `.renderer` function, which is encapsulated by the `guard` directive in Lit.
The examples do so for performance reasons.
For more details on the `guard` directive, refer to the https://lit.dev/docs/templates/directives/#guard[official Lit documentation].
```
